### DISCUSSION TITLE
HTTP Metrics Missing from Keycloak Metrics Endpoint

### DISCUSSION THREAD

**Discussion by parmeshwar11** [keycloak/keycloak#32769] [Q&A]
I`m working with Keycloak and have successfully enabled metrics using the CLI option --metrics-enabled and the environment variable KC_METRICS_ENABLED. Metrics are visible at the /auth/metrics endpoint. However, I'm noticing that HTTP-specific metrics, such as request counts, response times, and status codes, are missing from this endpoint.

Keycloak version: 22.0.5

Is there any additional configuration needed to include HTTP-specific metrics in Keycloak? Are there any known issues or additional steps required to expose these HTTP metrics? Any guidance or suggestions would be greatly appreciated!

**Comment by sschu** [comment]
There is --http-metrics-histograms-enabled, see https://www.keycloak.org/server/all-config.

**Reply by ahus1** [reply]
Please upgrade to KC25 and the HTTP metrics will be enabled by default. They are not supported in earlier versions.
